### ISSUE TITLE
not installable via pip install on python 3.11

### ISSUE THREAD

**Issue by yarikoptic** [Aarhus-Psychiatry-Research/timeseriesflattener#135]
#133 might be related since 3.11 is not tested

```
❯ py=3.11; d=venvs/dev$py; python$py -m venv $d && source $d/bin/activate && pip install timeseriesflattener
Collecting timeseriesflattener
  Downloading timeseriesflattener-0.23.2-py3-none-any.whl (36 kB)
Collecting SQLAlchemy<1.5.42,>=1.4.41
  Downloading SQLAlchemy-1.4.46-cp311-cp311-manylinux_2_5_x86_64.manylinux1_x86_64.manylinux_2_17_x86_64.manylinux2014_x86_64.whl (1.6 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 1.6/1.6 MB 4.7 MB/s eta 0:00:00
Collecting jupyter<1.1.0,>=1.0.0
  Using cached jupyter-1.0.0-py2.py3-none-any.whl (2.7 kB)
Collecting pyarrow<11.1.0,>=9.0.0
  Downloading pyarrow-11.0.0-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (34.9 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 34.9/34.9 MB 5.3 MB/s eta 0:00:00
Collecting dill<0.3.7,>=0.3.0
  Using cached dill-0.3.6-py3-none-any.whl (110 kB)
Collecting scipy<1.9.4,>=1.8.0
  Downloading scipy-1.9.3-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (33.4 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 33.4/33.4 MB 5.7 MB/s eta 0:00:00
Collecting psutil<6.0.0,>=5.9.1
  Using cached psutil-5.9.4-cp36-abi3-manylinux_2_12_x86_64.manylinux2010_x86_64.manylinux_2_17_x86_64.manylinux2014_x86_64.whl (280 kB)
Collecting pyodbc<4.0.36,>=4.0.34
  Downloading pyodbc-4.0.35-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (343 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 344.0/344.0 kB 5.4 MB/s eta 0:00:00
Collecting wasabi<1.1.2,>=0.9.1
  Downloading wasabi-1.1.1-py3-none-any.whl (27 kB)
Collecting dask<2023.2.0,>=2022.9.0
  Downloading dask-2023.1.1-py3-none-any.whl (1.1 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 1.1/1.1 MB 5.5 MB/s eta 0:00:00
Collecting protobuf<=3.20.3
  Downloading protobuf-3.20.3-py2.py3-none-any.whl (162 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 162.1/162.1 kB 7.8 MB/s eta 0:00:00
Collecting wandb<0.13.8,>=0.12.0
  Downloading wandb-0.13.7-py2.py3-none-any.whl (1.9 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 1.9/1.9 MB 5.4 MB/s eta 0:00:00
Collecting srsly<2.4.6,>=2.4.4
  Downloading srsly-2.4.5-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (489 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 489.2/489.2 kB 5.2 MB/s eta 0:00:00
Collecting skimpy<0.1.0,>=0.0.7
  Downloading skimpy-0.0.8-py3-none-any.whl (14 kB)
Requirement already satisfied: pandas<1.6.0,>=1.4.0 in ./venvs/dev3.11/lib/python3.11/site-packages (from timeseriesflattener) (1.5.3)
Collecting catalogue<2.1.0,>=2.0.0
  Downloading catalogue-2.0.8-py3-none-any.whl (17 kB)
Collecting pydantic<1.10.0,>=1.9.0
  Downloading pydantic-1.9.2-py3-none-any.whl (143 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 143.3/143.3 kB 5.3 MB/s eta 0:00:00
Requirement already satisfied: numpy<1.24.2,>=1.23.3 in ./venvs/dev3.11/lib/python3.11/site-packages (from timeseriesflattener) (1.23.5)
Collecting coloredlogs<15.1.0,>14.0.0
  Downloading coloredlogs-15.0.1-py2.py3-none-any.whl (46 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 46.0/46.0 kB 5.3 MB/s eta 0:00:00
Collecting scikit-learn<1.2.2,>=1.1.2
  Downloading scikit_learn-1.2.1-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (9.6 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 9.6/9.6 MB 6.2 MB/s eta 0:00:00
Collecting deepchecks<0.12.0,>=0.8.0
  Downloading deepchecks-0.11.1-py3-none-any.whl (7.6 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 7.6/7.6 MB 6.3 MB/s eta 0:00:00
Collecting timeseriesflattener
  Downloading timeseriesflattener-0.23.1-py3-none-any.whl (36 kB)
  Downloading timeseriesflattener-0.23.0-py3-none-any.whl (36 kB)
ERROR: Cannot install timeseriesflattener==0.23.0, timeseriesflattener==0.23.1 and timeseriesflattener==0.23.2 because these package versions have conflicting dependencies.

The conflict is caused by:
    timeseriesflattener 0.23.2 depends on psycopmlutils<0.4.0 and >=0.2.4
    timeseriesflattener 0.23.1 depends on psycopmlutils<0.4.0 and >=0.2.4
    timeseriesflattener 0.23.0 depends on psycopmlutils<0.4.0 and >=0.2.4

To fix this you could try to:
1. loosen the range of package versions you've specified
2. remove package versions to allow pip attempt to solve the dependency conflict

ERROR: ResolutionImpossible: for help visit https://pip.pypa.io/en/latest/topics/dependency-resolution/#dealing-with-dependency-conflicts
```

overall, if possible, I would strongly advise to strive to avoid upper version bounds since they are the ones which usually cause such difficulties. Testing and CI can ensure that you remain compatible.

**Comment by MartinBernstorff** [comment]
Ah yeah, we definitely had problems with 3.11 - we use dependabot and automerge to keep dependencies up to date, but obviously didn't test for this case. Expanding test cases to include 3.11 now.

**Comment by MartinBernstorff** [comment]
Fixed the `psycopmlutils` dependency, revealed another dependency that did not support Python 3.11. 

For more, see: https://github.com/Aarhus-Psychiatry-Research/timeseriesflattener/pull/136#issuecomment-1463808582

Expanded the python range as much as possible in https://github.com/Aarhus-Psychiatry-Research/timeseriesflattener/pull/136

Please let me know if you have other suggestions :-)

**Comment by MartinBernstorff** [comment]
Fixed in #153.
